channels/chan_voter.c

```diff
@@ -4705,8 +4704,33 @@ static void *voter_reader(void *data)
 							}
 							ast_copy_string(client->saved_challenge, proxy.challenge, sizeof(client->saved_challenge) - 1);
 							client->proxy_sin = psin;
+							/* Is the mix mode flag being sent by the proxy client? */
 							if (proxy.flags & 32) {
-								client->mix = 1;
+								/* The CLIENT has to send us flags to tell us it is configured for mix mode (GPS PPS = NONE)
+								 * so this is where we check the flags from the client, and update client->mix accordingly.
+								 * Mix mode requires a buflen >= 160 in voter.conf, which is equivalent to client->buflen = 1280
+								 * (buflen * 8). If a client connects as mix mode, we need to enforce the minimum buflen,
+								 * otherwie the client will connect, but cannot send us audio because the buffer isn't big enough.
+								 *
+								 * Check the buflen, throw an error if it is too small, and block the client from connecting.
+								 */
+								if (client->buflen < 1280) {
```

Comment:
Should we make this "magic number" a #define, and "maybe" relate it to the 160ms? (either do the math to create 1280 in the define, the other way aorund? )